packages/validator/src/validator.js

```diff
@@ -0,0 +1,166 @@
+);
+
+const checkMinValue: Validator = (minValue) => R.ifElse(
+  R.lte(minValue),
+  R.always(undefined),
+  R.always(ERROR.MIN_VALUE),
+);
+
+const checkMaxValue: Validator = (maxValue) => R.ifElse(
+  R.gte(maxValue),
+  R.always(undefined),
+  R.always(ERROR.MAX_VALUE),
+);
+
+const checkMaxFieldSize: Validator = (maxFieldSize) => R.ifElse(
+  R.pipe(
+    R.length,
+    R.gte(maxFieldSize),
+  ),
+  R.always(undefined),
+  R.always(ERROR.FIELD_SIZE),
+);
+
+const checkTextFormat: Validator = (textFormat) => R.ifElse(
+  R.both(
+    R.complement(R.isNil),
+    R.test(TEXT_FORMAT_PATTERN[textFormat]),
+  ),
+  R.always(undefined),
+  R.always(ERROR[textFormat]),
+);
+
+const checkDateFormat: Validator = (dateFormat) => R.ifElse(
+  R.both(
+    R.complement(R.isNil),
+    R.test(DATE_FORMAT_PATTERN[dateFormat]),
+  ),
+  R.always(undefined),
+  R.always(ERROR[dateFormat]),
+);
+
+const getCommonValidators: ValidatorsGetter = (field) => {
+  const validators = [];
+
+  if (field.isRequired) {
+    validators.push(checkRequired);
+  }
+
+  return validators;
+};
+
+const getNumberFieldValidators: ValidatorsGetter = (field) => {
+  const validators = [
+    ...getCommonValidators(field),
+  ];
+
+  const maxPrecision = getMaxPrecision(field);
+  const minValue = getMinValue(field);
+  const maxValue = getMaxValue(field);
+
+  if (maxPrecision) {
+    validators.push(checkMaxPrecision(maxPrecision));
+  }
+
+  if (minValue) {
+    validators.push(checkMinValue(minValue));
+  }
+
+  if (maxValue) {
+    validators.push(checkMaxValue(maxValue));
+  }
+
+  return validators;
+};
+
+const getTextFieldValidators: ValidatorsGetter = (field) => {
+  const validators = [
+    ...getCommonValidators(field),
+  ];
+
+  const maxFieldSize: ?string = getMaxFieldSize(field);
+  const format = getFormat(field);
+
+  if (maxFieldSize) {
+    validators.push(checkMaxFieldSize(maxFieldSize));
+  }
+
+  if (format && TEXT_FORMAT_PATTERN[format]) {
+    validators.push(checkTextFormat(format));
+  }
+
+  return validators;
+};
+
+const getDateFieldValidators: ValidatorsGetter = (field) => {
+  const validators = [
+    ...getCommonValidators(field),
+  ];
+
+  const format = getFormat(field);
+
+  if (format) {
+    validators.push(checkDateFormat(format));
+  }
+
+  return validators;
+};
+
+const processValidators: ValidatorsProcesser = validators => value => R.pipe(
+  R.map(validator => validator(value)),
+  R.find(R.is(String)),
+)(validators);
+
+/**
+ * ValidatorFacade creates validaton function based on field metadata
+ * @param field - Field metadata
+ * @returns {Function} - Validation function
+ */
+const validatorFacade: (field: Field) => (value?: any) => ?ValidationError = R.pipe(
+  R.cond([
+    [R.propEq('fieldType', FIELD_TYPE.NUMBER), getNumberFieldValidators],
+    [R.propEq('fieldType', FIELD_TYPE.TEXT), getTextFieldValidators],
+    [R.propEq('fieldType', FIELD_TYPE.DATE), getDateFieldValidators],
+    [R.T, getCommonValidators],
```

Comment:
@vorobeez what are you think about throwing error when we receive field with unsupported field type?

Reply:
Will be implemented.